ibis/file/parquet.py

```diff
@@ -0,0 +1,132 @@
+import ibis
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+from ibis.file.client import FileClient
+from ibis.pandas.core import pre_execute, execute
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+
+def connect(dictionary):
+        return ParquetClient(dictionary)
+
+
+_ARROW_DTYPE_TO_IBIS_TYPE = {
+    'int8': dt.int8,
+    'int16': dt.int16,
+    'int32': dt.int32,
+    'int64': dt.int64,
+    'uint8': dt.uint8,
+    'uint16': dt.uint16,
+    'uint32': dt.uint32,
+    'uint64': dt.uint64,
+    'halffloat': dt.float16,
+    'float': dt.float32,
+    'double': dt.float64,
+    'string': dt.string,
+    'timestamp[ns]': dt.timestamp,
+}
+
+_PARQUET_DTYPE_TO_IBIS_TYPE = {
+    'UTF8': dt.string,
+    'BYTE_ARRAY': dt.binary,
+    'TIMESTAMP_MICROS': dt.timestamp,
+    'FLOAT': dt.float32,
+    'DOUBLE': dt.float64,
+    'BOOLEAN': dt.boolean,
+    'INT32': dt.int32,
+    'INT64': dt.int64,
+    'UINT_8': dt.uint8,
+    'UINT_16': dt.uint16,
+    'UINT_32': dt.uint32,
+    'UINT_64': dt.uint64,
+}
+
+
+def arrow_types_to_ibis_schema(schema):
+    pairs = []
+    for cs in schema:
+        column_name = cs.name
+        ibis_type = _ARROW_DTYPE_TO_IBIS_TYPE[str(cs.type)]
+        pairs.append((column_name, ibis_type))
+    return ibis.schema(pairs)
+
+
+def parquet_types_to_ibis_schema(schema):
+    pairs = []
+    for cs in schema:
```

Comment:
@wesm is there (maybe internal) function to convert a ParquetColumnSchema to an Arrow schema?
would prefer to not have these parquet routines directly exposed here, rather have arrow types (which trivially convert to ibis types)

Reply:
The code exists, but needs to be exposed: https://issues.apache.org/jira/browse/ARROW-1759